illumos-utils/src/zfs.rs

```diff
@@ -137,15 +137,15 @@ impl Zfs {
     }
 
     /// Creates a new ZFS filesystem named `name`, unless one already exists.
-    pub fn ensure_zoned_filesystem(
+    pub fn ensure_filesystem(
```

Comment:
When provisioning filesystems in `sled-hardware/src/disk.rs` this doesn't want `zoned`, so it's configurable.

This function was also renamed to make the intent a bit more clear.